sky/specs/style-guide.md

```diff
@@ -8,44 +7,114 @@ that everyone, whether reading the code for the first time or
+
+Example:
+
+```dart
+// BAD:
+
+// What should this be?
+
+// This is a workaround.
+
+
+// GOOD:
+
+// According to this specification, this should be 2.0, but according to that
+// specification, it should be 3.0. We split the difference and went with
+// 2.5, because we didn't know what else to do.
+
+// TODO: Converting color to RGB because class Color doesn't support hex yet.
+//       See http://link/to/a/bug/123
+```
+
+## Dart
 
 In general, follow the [Dart style
-guide](https://www.dartlang.org/articles/style-guide/) for Dart code,
-except where that would contradict this page. (`dartfmt` [doesn't yet
-support](https://github.com/dart-lang/dart_style/issues/442) our style
-rules, so avoid using that for now.)
+guide](https://www.dartlang.org/articles/style-guide/) for Dart code, except
+where that would contradict this page.
 
-Always use the Dart Analyzer. Avoid checking in code that increases
-the output of the analyzer unless you've filed a bug with the Dart
-team. (Use `flutter analyze` to run the analyzer on Flutter code.)
 
-Use assert()s liberally to describe the contracts that you expect your
-code to follow.
+### DON'T use dartfmt (yet)
 
+Flutter code tends to use patterns that the standard `dartfmt` does not handle
+well. We are
+[working with the Dart team](https://github.com/dart-lang/dart_style/issues/442)
+to make dartfmt aware of these patterns.
 
-Types (i.e. classes, typedefs (function signature definitions) and
-enums) are named UpperCamelCase. Everything else (methods, fields,
-variables, constants, enum values, etc) is lowerCamelCase. Constant
-doubles and strings are prefixed with k. Prefer using a local const
-or a static const in a relevant class than using a global constant.
+
+### DO use the Dart Analyzer before submitting code
+
+While editing code Atom's `dartlang` plugin runs the analyzer automatically,
+preventing surprises later when you need to submit the code.
+
+Run `flutter analyzer --flutter-repo` prior to submitting your code for review.
+
+Avoid checking in code that increases the output of the analyzer. If a warning
+must be allowed due to a bug in the analyzer file a bug with the Dart team at
+http://dartbug.com/new.
+
+
+### PREFER using asserts to enforce contracts and invariants
+
+`assert()` allows us to be diligent about correctness without paying a
+performance penalty in release mode, because Dart only evaluates asserts in
+checked mode.
+
+Examples:
+
+```dart
+// A simple assertion.
+void paint() {
+  assert(needsPaint);
+}
+
+// A more complex assertion can be wrapped in a closure.
+class UnitCircle {
+  UnitCircle.fromVectors(double centerX, double centerY, double x, double y) {
+    assert(() {
+      double deltaX = x - centerX;
+      double deltaY = y - centerY;
+      const double epsilon = 0.00001;
+      return (math.pow(deltaX) + math.pow(deltaY) - 1.0).abs() < epsilon;
+    });
+  }
+}
+```
+
+
+### DO camel-case constant doubles and strings, and begin with prefix "k"
```

Comment:
This is grammatically dubious too. I think this would end up being more readable if we used topic headings (e.g. "Identifiers") and then described the style in full sentences rather than trying to shoe-horn the style description into the style used by the dart style guide.